homeassistant/components/obihai/__init__.py

```diff
@@ -32,7 +32,7 @@ async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         new_unique_id = await hass.async_add_executor_job(
             requester.pyobihai.get_device_mac
         )
-        hass.config_entries.async_update_entry(entry, unique_id=new_unique_id)
+        hass.config_entries.async_update_entry(entry, unique_id=new_unique_id.lower())
```

Comment:
You could use `format_mac` instead, to be consistent with other components.
`from homeassistant.helpers.device_registry import format_mac`

I would also suggest renaming the `new_unique_id` variable to `mac` or `device_mac` as it's not a unique_id until assigned to the config entry.

```suggestion
        hass.config_entries.async_update_entry(entry, unique_id=format_mac(mac))
```